### PR TITLE
Use async function instead of promise (#325)

### PR DESCRIPTION
**Pull request by weihanglo**
Changed all promises into async functions for express.

I have attempted to promsify all Redis client methods as #64 described. It failed but I have no idea why, and I cannot log directly in [storage.js](https://github.com/mozilla/send/blob/424eb2c37ad50ab7778985edbea029e8683e4720/server/storage.js). Any help?

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -117,93 +117,79 @@ app.get('/jsconfig.js', (req, res) => {
...
-          });
-        });
-      });
-    })
-    .catch(() => {
-      res.status(404).render('notfound');
+  try {
+    const filename = await storage.filename(id);
+    const contentLength = await storage.length(id);
+    const timeToExpiry = storage.ttl(id);
```
Comment (dannycoates) [inline server/server.js]: this needs an `await`

Comment (weihanglo) [inline server/server.js]: Done 596ad87!

**Comment by dannycoates** [comment]
As you've discovered converting storage.js is going to be more complicated. Let's do that in a separate PR. The easiest strategy is probably to add bluebird, as recommended in the node_redis [readme](https://github.com/NodeRedis/node_redis#promises). We'll also probably need to adjust the code a bit if that changes the behavior or rejecting, i.e. change some try/catches into checking result values.

**Comment by dannycoates** [comment]
Thanks @weihanglo! I'll verify this and merge it in the morning :)

File: server/server.js [hunk 2]
```diff
@@ -218,15 +204,15 @@ app.post('/delete/:id', (req, res) => {
...
-    .delete(id, delete_token)
-    .then(err => {
-      if (!err) {
-        log.info('Deleted:', id);
-        res.sendStatus(200);
-      }
-    })
-    .catch(err => res.sendStatus(404));
+  try {
+   const err = await storage.delete(id, delete_token);
```
Comment (pdehaan) [inline server/server.js]: Super minor nit, but I think the indenting is slightly off here. Not sure if you want to try risking running `npm run format` and running the **prettier** formatter against the codebase.

File: server/server.js [hunk 3]
```diff
@@ -254,48 +240,50 @@ app.post('/upload', (req, res, next) => {
...
-      .forceDelete(newId)
-      .then(err => {
-        if (!err) {
-          log.info('Deleted:', newId);
-        }
-      })
-      .catch(err => {
-        log.info('DeleteError:', newId);
-      });
+  req.on('close', async err => {
```
Comment (pdehaan) [inline server/server.js]: Unrelated to PR, but I just noticed that the `.on('close')` may return an `err` response here (apparently), which we aren't checking, but then we're overwriting the `err` value below on line 266.  🤷‍♀️ 
Behavior looks the same as it was previously, but there may be dragons here. Very small, uninteresting dragons.

File: server/server.js (hunk 2 above)
Comment (weihanglo) [inline server/server.js]: Sorry I totally forgot it! Maybe we should add pre-commit or pre-push hook to prevent these kind of mistake.

**Comment by dannycoates** [comment]
Thanks again, @weihanglo!

**Pull request merged**